src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkShadowSettingsInspection.kt

```diff
@@ -21,12 +21,12 @@ class SmkShadowSettingsInspection : SnakemakeInspection()  {
                 return
             }
 
-            val stringLiteral = st.argumentList?.arguments?.get(0)
+            val argument = if (st.argumentList?.arguments?.size != 0) st.argumentList!!.arguments[0] else null
 
-            if (stringLiteral != null &&
-                    stringLiteral is PyStringLiteralExpression &&
-                    stringLiteral.stringValue !in ShadowSectionSettingsProvider.SHADOW_SETTINGS) {
-                registerProblem(stringLiteral.originalElement,
+            if (argument != null &&
+                    argument is PyStringLiteralExpression &&
```

Comment:
I assume that it is false that `null is PyStringLiteralExpression` because `PyStringLiteralExpression` is not-null type, so you could simplify the condition, but please check it before.